server/core/runtime/runtime.go

```diff
@@ -93,7 +94,7 @@ func GetPlanFilename(workspace string, projName string) string {
 		return fmt.Sprintf("%s.tfplan", workspace)
 	}
 	projName = strings.Replace(projName, "/", planfileSlashReplace, -1)
-	return fmt.Sprintf("%s-%s.tfplan", projName, workspace)
+	return fmt.Sprintf("%s::%s.tfplan", projName, workspace)
```

Comment:
Can we leave this as a `-` ?

Reply:
If we leave this as `-` we are going to have issues when trying to get the project name. Before this was easy because we could get the last part of the cloned directory and `substring` from the plan file, but now we have to discover the workspace and the project name from the file because we no longer clone using workspace.

Picture this, the user tries to run a plan on the project `my-project` on the workspace `my-workspace`, given that we don't know either the project name nor the workspace how should we get the workspace and the project name from the file `my-project-my-workspace.tfplan`?

If we split by `-` we can have the following combinations:
- project: `my`, workspace: `project-my-workspace`
- project: `my-project`, workspace: `my-workspace`
- project: `my-project-my`, workspace: `workspace`

`::` was the safest separator I could find, since looks like users can't use it on the config files.

Reply:
It seems like the pr is trying to derive information from the file name and it shouldn't. The file name should be stateless even though it has this metadata baked in.

Can we derive the workspace and other metadata in a more procedural way than parsing the file name?

Reply:
cc @jamengual @krrrr38 @chenrui333 for more visibility

Reply:
It may have always been this way, but I do not think we should introduce a new level of complexity in order to continue down this path.

If using the `-` is too confusing due to the fact that we depend on the plan file name to parse out metadata, then instead of using `::` as a delimiter, then perhaps we should not be parsing metadata from the plan file, and instead look for a better method.

Reply:
I agree that getting information from the file name is not the best scenario, maybe we can get that information from the Backend interface. I will test this real quick to see if is possible.

Reply:
Ok, I've managed to get the information from the Backend interface on `pending_plan_finder.go@findWithAbsPaths()`, with that `ProjectNameFromPlanfile` and `WorkspaceNameFromPlanfile` are no longer necessary, I just need to adjust the tests and revert everything back to `-`.

The code looks like this:
```
func (p *DefaultPendingPlanFinder) findWithAbsPaths(pullRequest models.PullRequest) ([]PendingPlan, []string, error) {
	var plans []PendingPlan
	var absPaths []string

	pullDir, err := p.WorkingDir.GetPullDir(pullRequest.BaseRepo, pullRequest)
	if err != nil {
		return nil, nil, err
	}

	status, err := p.Backend.GetPullStatus(pullRequest)
	if err != nil {
		return nil, nil, err
	}

	for _, project := range status.Projects {
		absPath := fmt.Sprintf("%s/%s", pullDir, project.RepoRelDir)

		plans = append(plans, PendingPlan{
			RepoDir:     pullDir,
			RepoRelDir:  fmt.Sprintf("%s/%s", pullDir, project.RepoRelDir),
			Workspace:   project.Workspace,
			ProjectName: project.ProjectName,
		})
		absPaths = append(absPaths, absPath)
	}

	return plans, absPaths, nil
}
```

Reply:
Ok, tests adjusted, it was not easy, a lot of stuff seems to depend on the `pending_plan_finder`, the code still not good enough though and I need to validate more stuff before making sure everything is working.